velox/functions/sparksql/String.h

```diff
@@ -455,4 +455,83 @@ struct SubstrFunction {
   }
 };
 
+template <typename T>
+struct OverlayFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  // ASCII input always produces ASCII result.
+  static constexpr bool is_default_ascii_behavior = true;
```

Comment:
Consider also setting reuse_strings_from_arg to avoid copying strings. See https://facebookincubator.github.io/velox/develop/scalar-functions.html#zero-copy-string-result

Reply:
Thanks for your suggestions, I'm confused about reuse_strings_from_arg with these questions:

1, Could `reuse_strings_from_arg` be worked with multi input string? such as `func(out_type<Varchar>& result, const arg_type<Varchar>& inpu1, const arg_type<Varchar>& intput2)`, result will construct from parts of input1 and parts of input2.

2, If answer of question 1 is yes, how to append more than one strings with StringWriter when setting `reuse_strings_from_arg`? StringWriter::append or StringWriter::setNoCopy?

Reply:
Sorry, I got confused myself. Was seeing a lot of substring logic and forgot that this function is not a substring.